runtime/turing/src/lib.rs

```diff
@@ -1180,6 +1197,39 @@ impl_runtime_apis! {
 		}
 	}
 
+	impl pallet_xcmp_handler_rpc_runtime_api::XcmpHandlerApi<Block, Balance> for Runtime {
+		fn parachain_id() -> u32 {
+			ParachainInfo::parachain_id().into()
+		}
+
+		fn fees(encoded_xt: Bytes) -> Balance {
+			let extrinsic: <Block as BlockT>::Extrinsic = Decode::decode(&mut &*encoded_xt).unwrap();
+			if let Call::AutomationTime(pallet_automation_time::Call::schedule_xcmp_task{
+				provided_id, execution_times, para_id, currency_id, encoded_call, encoded_call_weight
+			}) = extrinsic.clone().function {
+				let len = encoded_xt.len() as u32;
+
+				let xcm_fee = XcmpHandler::calculate_xcm_fee_and_weight(
+					u32::from(para_id),
+					CurrencyId::from(currency_id),
+					encoded_call_weight,
+				).unwrap().0;
```

Comment:
is this the right way to handle potential errors for RPCs?

Reply:
@rhuttman Definitely not and I was hoping you had a better solution.  My last RPC implemented resolved this in a more user friendly way (https://github.com/OAK-Foundation/OAK-blockchain/blob/master/runtime/turing/src/lib.rs#L1206). It didn't feel like a very good approach either.

Reply:
I would like a similar thing here is correct. If we do not support that chain/currency combo then we should ultimately return an error. I would think this fcn returns a result and the RPC fcn that calls this wraps that into a runtime error w/ a special message